### ISSUE TITLE
SA1012 incorrectly reported on opening brace in property pattern nested in C# 11 list pattern

### ISSUE THREAD

**Issue by jnm2** [DotNetAnalyzers/StyleCopAnalyzers#3509]
Same setup as https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3503.

```cs
// ⚠ SA1012 Opening brace should be preceded by a space
//                    ↓
if (new[] { "a" } is [{ Length: 1 }])
{
}
```
